src/core/ZkitTSGenerator.ts

```diff
@@ -37,6 +39,23 @@ export default class ZkitTSGenerator extends BaseTSGenerator {
         circuitClasses.push({
           name: this._getCircuitName(artifacts[0].circuitArtifact),
           object: this._getCircuitName(artifacts[0].circuitArtifact),
+          protocol: artifacts[0].protocol,
+        });
+
+        continue;
+      }
+
+      if (artifacts.length === 2 && artifacts[0].protocol !== artifacts[1].protocol) {
+        circuitClasses.push({
+          name: this._getCircuitName(artifacts[0].circuitArtifact),
+          object: this._getCircuitName(artifacts[0].circuitArtifact) + this._getPrefix(artifacts[0].protocol!),
+          protocol: artifacts[0].protocol,
+        });
+
+        circuitClasses.push({
+          name: this._getCircuitName(artifacts[1].circuitArtifact),
+          object: this._getCircuitName(artifacts[1].circuitArtifact) + this._getPrefix(artifacts[1].protocol!),
+          protocol: artifacts[1].protocol,
```

Comment:
let keys = [...new Set(Object.keys(circuits))];

...

```suggestion
      if (artifacts.length > 1) {
        for (let j = 0; j < artifacts.length; j++) {  
          circuitClasses.push({
            name: this._getCircuitName(artifacts[j].circuitArtifact),
            object: this._getCircuitName(artifacts[j].circuitArtifact) + this._getPrefix(artifacts[j].protocol!),
            protocol: artifacts[j].protocol,
          });
        }
```